components/user-role-management.tsx

```
@@ -48,61 +61,64 @@ const demoUsers = [
-  const { session, setRole } = useDemoAuth()
-  const [isUpdating, setIsUpdating] = useState<Record<string, boolean>>({})
+export function UserRoleManagement({
+  users = demoUsers,
+}: UserRoleManagementProps) {
+  const router = useRouter();
+  const { session, setRole } = useDemoAuth();
+  const [isUpdating, setIsUpdating] = useState<Record<string, boolean>>({});
   const [selectedRoles, setSelectedRoles] = useState<Record<string, string>>(
-    users.reduce((acc, user) => ({ ...acc, [user.id]: user.role }), {}),
-  )
+    users.reduce((acc, user) => ({ ...acc, [user.id]: user.role }), {})
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)